### ISSUE TITLE
VMware: ui error: Build 'X' errored: lstat output-X: no such file or directory

### ISSUE THREAD

**Issue by maxx-bachurin** [mitchellh/packer#3284]
Have encountered the following problem when building a vm:

```
2016/02/29 11:25:28 ui: ==>  FusionUI: Unregistering virtual machine...
==>  FusionUI: Unregistering virtual machine...
2016/02/29 11:25:28 packer: 2016/02/29 11:25:28 opening new ssh session
2016/02/29 11:25:28 packer: 2016/02/29 11:25:28 starting remote command: vim-cmd vmsvc/unregister 36
2016/02/29 11:25:29 packer: 2016/02/29 11:25:29 remote command exited with '0': vim-cmd vmsvc/unregister 36
2016/02/29 11:25:29 ui error: Build ' FusionUI' errored: lstat output-FusionUI: no such file or directory
2016/02/29 11:25:29 Builds completed. Waiting on interrupt barrier...
Build ' FusionUI' errored: lstat output-FusionUI: no such file or directory
2016/02/29 11:25:29 machine readable: error-count []string{"1"}
2016/02/29 11:25:29 ui error:
==> Some builds didn't complete successfully and had errors:
2016/02/29 11:25:29 machine readable:  FusionUI,error []string{"lstat output-FusionUI: no such file or directory"}
2016/02/29 11:25:29 ui error: -->  FusionUI: lstat output-FusionUI: no such file or directory
2016/02/29 11:25:29 ui:
==> Builds finished but no artifacts were created.
2016/02/29 11:25:29 waiting for all plugin processes to complete...

==> Some builds didn't complete successfully and had errors:
-->  FusionUI: lstat output-FusionUI: no such file or directory

==> Builds finished but no artifacts were created.
2016/02/29 11:25:29 /home/test/packer: plugin process exited
2016/02/29 11:25:29 [ERR] Error decoding response stream 7: EOF
2016/02/29 11:25:29 /home/test/packer: plugin process exited
```

When 'output-FusionUI' is present on the datastorage and contains the vm
- Packer Version
  Packer v0.9.0
- Host platform
  VMWare ESXi 6.0.0 build 3029758
- The template and scripts
  https://gist.github.com/maxx-bachurin/2e3e7544b0508f7900c0

the script
```
# !/bin/bash

echo 'xxxx' | sudo -S apt-get update -y
echo 'xxxx' | sudo -S apt-get upgrade -y
echo 'xxxx' | sudo -S reboot
sleep 60
```

**Comment by rickard-von-essen** [comment]
Can you rerun with packer log ```PACKER_LOG=1 packer build ...``` and gist the log.

**Comment by maxx-bachurin** [comment]
here you go https://gist.github.com/maxx-bachurin/61f1922c609b065425db

**Comment by rickard-von-essen** [comment]
Could you try rerunning without ```vm_name``` and ```name``` and ```vmdk_name``` set? (Also adjust the ```vmx``` map)

**Comment by maxx-bachurin** [comment]
I will try. What do you mean by "Also adjust the vmx map"?

**Comment by majinjing3** [comment]
@rickard-von-essen 
I have ran without vm_name and name and vmdk_name set, the same error

**Comment by majinjing3** [comment]
@maxx-bachurin @rickard-von-essen 
When I created a folder named "output-X" in current working dir. All is okay.
I think it maybe a bug.

**Comment by rickard-von-essen** [comment]
@majinjing3 Excellent information, it will help me to track down the code path with the problem. Currently I don't have any environment setup to test this. Maybe next week, but I can't make any promises.

**Comment by maxx-bachurin** [comment]
I can confirm the error was reproduced with adjusted config (as was suggested remove vm_name and name and vmdk_name)

**Comment by majinjing3** [comment]
@rickard-von-essen 
This PR :#3050  has solved  this bug.
@maxx-bachurin 
you can a format parameter, then this error will disappear.

**Comment by maxx-bachurin** [comment]
@majinjing3 yes, it seems you are right, I tested the line with the fix from #3050 on my copy of the packer sources. No issue is reproduced when using my packer build with the fix from #3050

**Comment by maxx-bachurin** [comment]
@majinjing3 do you mean to make use of a format parameter?
